src/sentry/static/sentry/app/components/acl/feature.jsx

```diff
@@ -95,31 +121,52 @@ class Feature extends React.Component {
       organization.includes(feature) ||
       project.includes(feature)
     );
-  };
+  }
 
   render() {
-    let {children, features, renderNoFeatureMessage, requireAll} = this.props;
+    let {
+      children,
+      features,
+      renderDisabled,
+      organization,
+      project,
+      requireAll,
+    } = this.props;
 
     let allFeatures = this.getAllFeatures();
     let method = requireAll ? 'every' : 'some';
     let hasFeature =
       !features || features[method](feat => this.hasFeature(feat, allFeatures));
 
+    // Default renderDisabled to the ComingSoon component
+    let customDisabledRender =
+      renderDisabled === false
+        ? false
+        : typeof renderDisabled === 'function' ? renderDisabled : () => <ComingSoon />;
+
+    // Override the renderDisabled function with a hook store function if there
+    // is one registered for the feature.
+    if (renderDisabled !== false && features.length === 1) {
+      HookStore.get(`feature-disabled:${descopeFeatureName(features[0])}`)
+        .slice(0, 1)
```

Comment:
I think we should attempt to render the entire array of hooks so that it's consistent with how we use hooks.

Reply:
I'll come back to this at some point here.